add pull down to update (issue 13) on schedule tables

sethlaw opened the pull request.

chrsmys commented on hackertracker/HTInitViewController.swift:
```diff
@@ -80,8 +80,8 @@ class HTInitViewController: UIViewController {
         let df = DateFormatter()
-        df.dateFormat = "yyyy-MM-dd HH:mm:ss"
-        df.timeZone = TimeZone(identifier: "PDT")
+        df.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSXXXXX"
```
Creating a date formatter is a fairly expensive operation. Can you add that to the `DateFormatterUtility` so that it is cached?

chrsmys commented on hackertracker/HTInitViewController.swift:
```diff
@@ -92,11 +92,11 @@ class HTInitViewController: UIViewController {
         let json = JSON(data: dataFromString!, options: JSONSerialization.ReadingOptions.mutableLeaves, error: nil)
         
-        let updateTime = json["updateTime"].string!
+        //let updateTime = json["updateTime"].string!
```
This should be removed.

chrsmys commented on hackertracker/HTScheduleTableViewController.swift:
```diff
@@ -12,13 +12,64 @@ import CoreData
+        request.httpMethod = "GET"
+        
+        var queue = OperationQueue()
+        var con = NSURLConnection(request: request as URLRequest, delegate: self, startImmediately: true)
```
NSURL Connection is fairly old. A newer class exists that makes this task much simpler. `NSURLSession` 

Here is an example

```
URLSession.shared.dataTask(with: request) { (data, response, error) in
  //ConnectionDidFinishLoading Logic here.
}
```

chrsmys commented on hackertracker/HTScheduleTableViewController.swift:
```diff
@@ -12,13 +12,64 @@ import CoreData
+
+    }
+    
+    func connection(_ con: NSURLConnection!, didReceiveData _data:Data!) {
```
If you move to URLSession you should no longer need this.

chrsmys commented on hackertracker/HTScheduleTableViewController.swift:
```diff
@@ -12,13 +12,64 @@ import CoreData
+        
+        let dataFromString = resStr!.data(using: String.Encoding.utf8.rawValue)
+        
+        let df = DateFormatter()
```
Can we move this to `DateFormatterUtility` if it doesn't already exist there

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+    
+    let json = JSON(data: data, options: JSONSerialization.ReadingOptions.mutableLeaves, error: nil)
+    
+    let df = DateFormatter()
```
`DateFormatterUtility`

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+        
+        var mySched : [Event] = []
+        
+        df.dateFormat = "yyyy-MM-dd HH:mm z"
```
We should have a separate date formatter entry in `DateFormatterUtility` for each format.

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+            if item["description"] != "" {
+                te.details = item["description"].string!
+            }
+            //NSLog("\(te.id): \(te.title) \(item["link"])")
```
This should be removed.

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+import CoreData
+
+
+func updateSchedule(_ data: Data) -> Bool {
```
We should look into how to break this into more modular functions.

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+            }
+            
+            te.who = item["who"].string!
+            var d = item["date"].string!
```
I think if we made the names date, begin, end it would be a lot more readable.

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+            var d = item["date"].string!
+            var b = item["begin"].string!
+            var e = item["end"].string!
+            if ( d == "" ) {
```
You could even do `if d.isEmpty {`

chrsmys commented on hackertracker/ScheduleHelper.swift:
```diff
@@ -0,0 +1,159 @@
+            if ( b != "" ) {
+                if ( b == "24:00") {
+                    b = "00:00"
+                    if ( d == "2016-08-04" ) {
```
What's with all of the 2016 dates?

chrsmys commented on hackertracker/ScheduleHelper.swift:
This would be a good candidate for unwrapping the optional with if let

```
if let error = err {
  print(error)
}
```

sethlaw commented:
@chrsmys, please review again and merge this in. Cleaned up dateformatters and switched to URLSession. The rest of this will get refactored when the new JSON is available.

chrsmys commented on hackertracker/HTInitViewController.swift:
```diff
@@ -113,7 +104,7 @@ class HTInitViewController: UIViewController {
         var mySched : [Event] = []
         
-        df.dateFormat = "yyyy-MM-dd HH:mm z"
+        //df.dateFormat = "yyyy-MM-dd HH:mm z"
```
We want to remove this comment
